src/test/java/com/spotify/dns/DnsSrvWatchersTest.java

```diff
@@ -0,0 +1,90 @@
+package com.spotify.dns;
+
+import com.google.common.collect.ImmutableList;
+
+import org.junit.Test;
+
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.not;
+
+public class DnsSrvWatchersTest {
+
+  @Test
+  public void triggerRace() throws Exception {
+    int limit = 100000;
+    while (limit-- > 0) {
+      System.out.println(limit);
+      resolve();
+    }
+  }
+
+  private void resolve() throws Exception {
+    final DnsSrvResolver srvResolver = new MockResolver(
+        "horse.sto3.spotify.net", LookupResult.create("localhost", 1, 0, 0, 0));
+
+    final AtomicReference<Set<LookupResult>> hosts = new AtomicReference<Set<LookupResult>>();
+    final CountDownLatch latch = new CountDownLatch(1);
+
+    final ChangeNotifier.Listener<LookupResult> listener = new MockListener(hosts, latch);
+
+    DnsSrvWatcher<LookupResult> watcher = DnsSrvWatchers.newBuilder(srvResolver)
+        .polling(1, TimeUnit.MILLISECONDS)
+        .build();
+    final ChangeNotifier<LookupResult> notifier = watcher.watch("horse.sto3.spotify.net");
+
+    // since I set fire to true I'd expect being notified either right away or after a millisecond
+    notifier.setListener(listener, true);
+    latch.await();
+    notifier.close();
+    watcher.close();
+    assertThat(hosts.get(), not(empty()));
```

Comment:
It would be slightly better to store the ```LookupResult``` used to instantiate the MockResolver and assert that the returned hosts is equal to a singleton list of that result.